cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -317,6 +317,58 @@ For example, give these arguments when starting the MapReduce through a RESTful
     "dataset.totals.output.partition.key.league" : "nfl"
   }
 
+Incrementally Processing Partitions of a PartitionedFileSets in MapReduce
+=========================================================================
+
+One way to process a partitioned file set is with a MapReduce program that runs repeatedly and,
+in every run, reads the partitions that have been added since its previous run. This requires
+that the MapReduce program persists between runs what partitions have already been consumed.
+An easy way to do that is the BatchPartitionConsumer, an experimental feature introduced in CDAP 3.1.0.
+The MapReduce program is responsible for extending this abstract consumer class with methods to
+persist and the read back its state. In the following example, the state is persisted to a row in a
```

Comment:
I took your suggestion here, except I left the 'also' after 'Datasets'. It seems like that emphasizes the fact that other datasets can be used in addition to KV Table.